CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration_TemplateMod.uc

```diff
@@ -1200,6 +1201,34 @@ static protected function string GetAcademySlotBonusDisplayString (XComGameState
 	return class'X2StrategyElement_DefaultStaffSlots'.static.GetBonusDisplayString(SlotState, "%SKILL", Contribution);
 }
 
+static function PatchCovertActionPromotionRewards()
+{
+	local X2StrategyElementTemplateManager TemplateManager;
+	local X2DataTemplate DataTemplate;
+	local X2CovertActionTemplate ActionTemplate;
+	local int index;
+	
+	TemplateManager = class'X2StrategyElementTemplateManager'.static.GetStrategyElementTemplateManager();
+
+	foreach TemplateManager.IterateTemplates(DataTemplate)
+	{
+		ActionTemplate = X2CovertActionTemplate(DataTemplate);
+
+		if (ActionTemplate != none)
+		{
+			// Must be a for loop because a foreach loop is passed a copy of the array
+			for (index = 0; index < ActionTemplate.Slots.Length; index++)
+			{
+				if (ActionTemplate.Slots[index].StaffSlot == 'CovertActionSoldierStaffSlot'
```

Comment:
What about `CovertActionRookieStaffSlot`? And any other slot template that custom CAs from other mods may use?

Reply:
Hmm, does that check even need to be there? As far as I can see, staff slots aren't handled differently so it's fine to just call RemoveItem on the empty rewards array

Reply:
Yeah, I think removing this check is probably the way to go